### PR TITLE
feat: Non-Linear font scaling

### PR DESCRIPTION
**Pull request by BrzozowskiPawel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [  ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³
----

With this PR, we fixed the problem of linear font scaling. Now it's non-linear and better represents each level of accessibility when the user wants to have bigger or smaller text. </br>
Implementation of this font also can't change the default app UI so the font should stay the same (without font size scaling) before and after this PR. </br>.

Photos (without scaling) of the app before implementation::
| settings | settings2 |  account | settings - advanced |
|---|---|---|---|
| Screenshot 2022-05-12 at 08 21 49  | Screenshot 2022-05-12 at 08 26 22  | Screenshot 2022-05-12 at 09 54 36 | Screenshot 2022-05-12 at 08 25 21  |

Photos of the app (with scaling) after implementation:
| settings | settings2 |  account | settings - advanced |
|---|---|---|---|
| Screenshot 2022-05-12 at 08 21 49  | Screenshot 2022-05-12 at 08 26 22  | Screenshot 2022-05-12 at 09 54 36 | Screenshot 2022-05-12 at 08 25 21  |

And here I present how scaling looks like: </br> </br>

Scaling before implementation:
| iPhone 8 - English  | iPhone 8 - German  |  iPhone SE - English | iPhone SE - German  |
|---|---|---|---|
| Screenshot 2022-05-12 at 08 21 49  | Screenshot 2022-05-12 at 08 26 22  | Screenshot 2022-05-12 at 09 54 36 | Screenshot 2022-05-12 at 08 25 21  |

Scaling after implementation:
| iPhone 8 - English  | iPhone 8 - German  |  iPhone SE - English | iPhone SE - German  |
|---|---|---|---|
| Screenshot 2022-05-12 at 08 21 49  | Screenshot 2022-05-12 at 08 26 22  | Screenshot 2022-05-12 at 09 54 36 | Screenshot 2022-05-12 at 08 25 21  |

### Issues

This PR fixes linear scaling and allows better customization.

### Solutions

This solution adds another function that returns a specific size for each FontSize and selected level of scaling.

Needs releases with:
None but for testing purpose it contains changes from [ACC-99](https://github.com/wireapp/wire-ios/pull/5675). 

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift [hunk 1]
```diff
@@ -44,11 +44,13 @@ class SettingsTableCell: UITableViewCell, SettingsCellType {
         return iconImageView
     }()
 
-    let cellNameLabel: UILabel = {
-        let label = DynamicFontLabel(fontSpec: .normalLightFont, color: .textForeground)
+    let cellNameLabel: DynamicFontLabel = {
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:47]: These changes seem unrelated to the PR.

File: Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift [hunk 2]
```diff
@@ -72,8 +74,10 @@ class SettingsTableCell: UITableViewCell, SettingsCellType {
     }()
 
     private let badgeLabel: UILabel = {
-        let badgeLabel = DynamicFontLabel(fontSpec: .smallMediumFont, color: .textInBadge)
+        let badgeLabel = UILabel()
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:77]: These changes seem unrelated to the PR.

Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:77]: Please revert those changes as well.

**Review by agisilaos — COMMENTED** [review]
I'm not so confident in merging this one in just yet. The changes here will impact most of the screens in the app and in the case of settings (as shown in the PR description) as you can see the text of navigation bar grows way more than it should. 

Any thoughts @johnxnguyen?

File: Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift (hunk 1 above)
Comment (BrzozowskiPawel) [reply Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:47]: Yes, i will chnage it back as it was

File: Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme [hunk 3]
```diff
@@ -199,11 +199,11 @@
          </CommandLineArgument>
          <CommandLineArgument
             argument = "-BackendEnvironmentTypeOverrideKey staging"
-            isEnabled = "NO">
+            isEnabled = "YES">
          </CommandLineArgument>
          <CommandLineArgument
             argument = "--persist-backend-type"
-            isEnabled = "NO">
+            isEnabled = "YES">
```
Comment (agisilaos) [inline Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme:206]: This needs to be reverted. We shouldn't commit those things.

**Comment by johnxnguyen** [comment]
>  as you can see the text of navigation bar grows way more than it should

@agisilaos indeed this is an issue, but it should be pretty easy to solve. I would suggest to create a separate text style for title bar titles, and then we can control how the grow / shrink independently of the other styles. This could be done on a separate branch.

**Review by agisilaos — APPROVED** [review]
